src/api/albums.js

```diff
@@ -1,37 +1,5 @@
 import baseURL from "./base_url";
 
-export function index(auth) {
```

Comment:
I would prefer if you just re-export the generic fetch method here with the first argument set. You can use a function reference in the store to still have a generic commit action.

Reply:
This is possible, but it feels weird to me. Do you still want this now that there is a generator function?

Reply:
> The store shouldn't need to know what the HTTP path is for this resource.

This is actually the main reason I want this. Seperation of concerns is important to me.

> Once we convert all the common api calls into generators, most of the files in /api would only serve to redefine functions.

I think only index can be generalized into a generator. I don't see how the other functions will provide intermediate results.

Reply:
> > The store shouldn't need to know what the HTTP path is for this resource.
> 
> This is actually the main reason I want this. Seperation of concerns is important to me.

But then couldn't we also define an object that defines the path for each collection? Then we could just tell the `indexGenerator` which collection it is dealing with

> > Once we convert all the common api calls into generators, most of the files in /api would only serve to redefine functions.
> 
> I think only index can be generalized into a generator. I don't see how the other functions will provide intermediate results.

You're right, i got confused. What I meant was that we could convert most of the api calls to a more generic functions. (Since often the `path` is the only thing that is different between all the actions of the same type).